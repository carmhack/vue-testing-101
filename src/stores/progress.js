import { defineStore } from "pinia";
import { computed, ref } from "vue";

export const useProgressStore = defineStore("progress", () => {
  const counter = ref(0);
  const max = ref(100);

  const increment = () => {
    counter.value += 1;
  }

  const percentage = computed(() => {
    return Math.round((counter.value / max.value) * 100);
  })

  return {
    counter,
    max,
    percentage,
    increment,
  }
})