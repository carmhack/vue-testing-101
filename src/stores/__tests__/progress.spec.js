import { createPinia, setActivePinia } from 'pinia';
import { describe, it, expect, beforeEach } from 'vitest'
import { useProgressStore } from '../progress';

describe('Progress Store', () => {
  let progress;

  beforeEach(() => {
    setActivePinia(createPinia());
    progress = useProgressStore();
  })

  it('starts at zero', () => {
    expect(progress.counter).toBe(0);
  })

  it('increments by 1', () => {
    progress.increment()
    expect(progress.counter).toBe(1);
    expect(progress.counter).toMatchSnapshot();
  })
  
  it('updates percentage', () => {
    progress.$patch({
      max: 10,
      counter: 5,
    })

    expect(progress.percentage).toBe(50);
  })
});