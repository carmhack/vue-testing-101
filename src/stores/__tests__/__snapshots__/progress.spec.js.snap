// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress Store > increments by 1 1`] = `1`;
